src/components/v2/V2Project/V2BalancesModal/V2BalancesModal.tsx

```diff
@@ -0,0 +1,171 @@
+      })),
+    })
+  }, [form, projectMetadata, editModalVisible])
+
+  async function updateTokenRefs() {
+    if (!projectName) return
+
+    await form.validateFields()
+
+    setLoading(true)
+
+    const editingTokenRefs: TokenRef[] = form
+      .getFieldsValue()
+      .tokenRefs.map(t => t.assetInput)
+      .map(a => ({ value: a.input, type: a.type }))
+
+    const uploadedMetadata = await uploadProjectMetadata(
+      {
+        ...projectMetadata,
+        tokens: editingTokenRefs.filter(t => t.type),
+      },
+      projectName,
+    )
+
+    if (!uploadedMetadata.IpfsHash) {
+      setLoading(false)
+      return
+    }
+
+    storeCidTx(
+      { cid: uploadedMetadata.IpfsHash },
+      {
+        onDone: () => {
+          setLoading(false)
+          setEditModalVisible(false)
+          form.resetFields()
+        },
+      },
+    )
+  }
+
+  return (
+    <Modal
+      visible={visible}
+      onCancel={onCancel}
+      footer={
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'space-between',
+            marginTop: 20,
+          }}
+        >
+          {hasEditPermissions ? (
+            <Button
+              type="text"
+              size="small"
+              icon={<SettingOutlined />}
+              onClick={() => setEditModalVisible(true)}
+            >
+              <span>
+                <Trans>Edit tracked assets</Trans>
+              </span>
+            </Button>
+          ) : (
+            <div></div>
+          )}
+          <Button onClick={onCancel}>
+            <Trans>Done</Trans>
+          </Button>
+        </div>
+      }
+    >
+      <div>
+        <h2>
+          <Trans>Assets</Trans>
+        </h2>
+        <p>
+          <Trans>Other assets in this project's owner's wallet.</Trans>
+        </p>
+
+        <Space direction="vertical" style={{ width: '100%', marginTop: 20 }}>
+          <V2ProjectTokenBalance projectId={V2_PROJECT_IDS.JUICEBOX_DAO} />
+
+          {(projectMetadata as ProjectMetadataV4)?.tokens?.map(t =>
+            t.type === 'erc20' ? (
+              <ERC20TokenBalance
+                key={t.value}
+                wallet={owner}
+                tokenAddress={t.value}
+              />
+            ) : (
+              <V2ProjectTokenBalance
+                projectId={BigNumber.from(t.value).toNumber()}
+              />
+            ),
+          )}
+        </Space>
+
+        <Modal
+          title={t`Edit tracked assets`}
+          visible={editModalVisible}
+          onCancel={() => setEditModalVisible(false)}
+          cancelText={t`Cancel`}
+          width={600}
+          confirmLoading={loading}
+          onOk={updateTokenRefs}
+          okText={t`Save tracked assets`}
+        >
+          <p style={{ marginBottom: 40 }}>
+            <Trans>
+              Display ERC-20 tokens and other Juicebox project tokens that are
+              in this project's owner's wallet.
```

Comment:
```suggestion
              Display ERC-20 and other Juicebox project tokens that this
              project owner's holds.
```

Reply:
should it be `that this project owner holds?` I think `onwer's holds` implies ownership of holds (noun) rather than verb(?) - im bad at grammar

Reply:
haha na that was my typo. `owner holds` ✅